lib/model/game_state.dart

```diff
@@ -14,23 +14,23 @@ import 'hand.dart';
 class GameState with ChangeNotifier {
   final Randomiser randomiser;
 
-  Card currentCard;
+  Card? currentCard;
   final List<Card> deck;
   final List<Card> discard = [];
   final List<Card> exile = [];
   final List<Card> scrying = [];
 
   final Hand playerHand = Hand();
 
-  GameResult result;
+  GameResult? result;
 
-  GameState(this.randomiser, this.deck, [List<Resource> initialResources]) {
-    playerHand.addResources(initialResources);
+  GameState(this.randomiser, this.deck, [List<Resource>? initialResources]) {
+    playerHand.addResources(initialResources!);
   }
 
   Card nextCard() {
     if (currentCard != null) {
-      discard.add(currentCard);
+      discard.add(currentCard!);
```

Comment:
This is a bit sad, I guess we still need the `!` in this case because `currentCard` isn't final and so in theory could be changed between the null check and the next line :thinking: 

Kotlin has a `.let` paradigm for this type of situation, where you'd write something like:

```
currentCard?.let {
    discard.add(it)
}
```

Here, `it` is the value that `currentCard` had at the time of the null-check, so is guaranteed to be non-null throughout the following closure. It's quite nice!

Sounds like Dart doesn't have an equivalent, but we might be able to arrange it via an extension function? https://stackoverflow.com/questions/52148319/what-is-darts-equivalent-to-kotlins-let